src/lint/mod.rs

```diff
@@ -263,8 +286,32 @@ pub mod tests {
         let example_path = format!("{}{}",
                                    env!("CARGO_MANIFEST_DIR"),
                                    EXAMPLE_CFG);
-        let example_cfg = parse_lint_cfg(example_path.into()).unwrap();
+        let (example_cfg, unknowns) = parse_lint_cfg(example_path.into()).unwrap();
         assert_eq!(example_cfg, LintCfg::default());
+        // Assert that there are no unknown fields in the example config:
+        assert!(unknowns.is_empty(), "Example config should not have unknown fields: {:?}", unknowns);
+    }
+
+    #[test]
+    fn test_unknown_fields_detection() {
+        use crate::lint::LintCfg;
+        
+        // JSON with unknown fields
+        let json_with_unknowns = r#"{
+            "sp_brace": {},
+            "unknown_field_1": true,
+            "indent_size": {"indentation_spaces": 4},
+            "another_unknown": "value"
+        }"#;
+        
+        let val: serde_json::Value = serde_json::from_str(json_with_unknowns).unwrap();
+        let mut unknowns = Vec::new();
+        let result = LintCfg::try_deserialize(&val, &mut unknowns);
+        
+        assert!(result.is_ok());
+        assert_eq!(unknowns.len(), 2);
+        assert!(unknowns.contains(&"unknown_field_1".to_string()));
+        assert!(unknowns.contains(&"another_unknown".to_string()));
     }
```

Comment:
It may be good to also assert the final LintCfg instance is equal to 
```
{
    "sp_brace": {},
    "indent_size": {"indentation_spaces": 4}
}
```